src/data/markdown/docs/03 cloud/03 Integrations/05 Cloud APM/03 Prometheus Remote Write.md

```diff
@@ -0,0 +1,122 @@
+| `includeTestRunId`      | If set, the `test_run_id` will be exported per each metric as an extra tag. Default is `false`.                                                                                                                                                                                               |
+
+## Supported Authentication Mechanisms
+
+Currently only bearer token and HTTP basic authentication mechanisms are supported. To configure your authentication mechanism, populate `credentials` with either of these configuration parameters:
+
+### HTTP basic authentication
+
+| Name       | Description                                |
+| ---------- | ------------------------------------------ |
+| `username` | The username in HTTP basic authentication. |
+| `password` | The password in HTTP basic authentication. |
+
+```javascript
+credentials: {
+    username: "<username>",
+    password: "<password>"
+}
+```
+
+### Bearer token authentication
+
+| Name    | Description                               |
+| ------- | ----------------------------------------- |
+| `token` | The bearer token without the prefix/type. |
+
+```javascript
+credentials: {
+    token: "<token>"
+}
+```
+
+## Example Configuration Object
+
+All the above configuration parameters are passed like this in your test run.
+
+```javascript
+export let options = {
+  ext: {
+    loadimpact: {
+      apm: [
+        {
+          provider: "prometheus",
+          remoteWriteURL: "<Remote Write URL>", // This can include query-string parameters
+          credentials: {
+              token: "<token>" // Optional
+          },
+          metrics: ["http_req_sending", "my_rate", "my_gauge", ...],
+          includeDefaultMetrics: true,
+          includeTestRunId: false
+        },
+      ]
+    },
+  },
+};
+```
+
+## Prometheus Remote Write Setup on New Relic
+
+For sending custom metrics from your test run to New Relic's Prometheus remote write integration, follow the [instructions](https://docs.newrelic.com/docs/integrations/prometheus-integrations/install-configure-remote-write/set-your-prometheus-remote-write-integration/) on their documentation.
+
+The `prometheus_server` parameter should be included in the `remoteWriteURL` configuration parameter. The bearer token can be included either as `credentials.token` (APM configuration parameter) or as part of the `remoteWriteURL` using the `X-License-Key` parameter, as mentioned in their documentation.
+
+So an example configuration for New Relic might look like this, if you use `X-License-Key` in your `remoteWriteURL`:
+
+```javascript
+export let options = {
+  ext: {
+    loadimpact: {
+      apm: [
+        {
+          provider: "prometheus",
+          remoteWriteURL: "https://metric-api.newrelic.com/prometheus/v1/write?X-License-Key=<YOUR_LICENSE_KEY>&prometheus_server=<YOUR_DATA_SOURCE_NAME>",
```

Comment:
I'm strongly against documenting how to set the license key as a queryParam in the URL. Do we **have** to have this due to compatibility with some of the major APMs? In that case, I fold. Otherwise, I insist that we scratch this example.

Reply:
It is because of New Relic, as stated in their [docs](https://docs.newrelic.com/docs/integrations/prometheus-integrations/install-configure-remote-write/set-your-prometheus-remote-write-integration/#setup) on how to setup the RW integration.

Reply:
Hey @gspncr, what do you think about this?

Also, I'd be very happy to have your review.

Reply:
hey both, sorry didn't get a notification. If you have the possibility to pass in Authorization header bearer token you could shape this in [the same way we have this documented for k8s and Helm](https://docs.datadoghq.com/integrations/mysql/?tab=host#data-collected), otherwise license key as a queryParam is the way to go

Reply:
@gspncr 
The link refers to DataDog.

Reply:
Major oops! [Updated the link](https://docs.newrelic.com/docs/integrations/prometheus-integrations/install-configure-remote-write/set-your-prometheus-remote-write-integration/#:~:text=kubernetes%20and%20helm%20remote%20write%20integrations) - it is just the k8s/Helm section of the docs you already were looking at 😉

Reply:
I think it is best to use one style, as @simskij mentioned. So, I removed the example that uses `LICENSE_KEY` in query params.